tgui/packages/tgui/interfaces/BarberingMenu.js

```diff
@@ -0,0 +1,119 @@
+  const { search_text, all_hair_names, all_hair_styles } = props;
+
+  const filtered_list = all_hair_names.filter((x) => x.toLowerCase().includes(search_text.toLowerCase()));
+  return filtered_list.map((value, index) => (<HairPreview
+    key={all_hair_styles[value].id}
+    hair_style={all_hair_styles[value]}
+    hair_name={value} />));
+};
+
+const HairOptions = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { selected_hair_portion } = data;
+
+  return (
+    <Stack justify="space-between" fontSize="15px">
+      <Button.Checkbox checked={selected_hair_portion === "bottom" ? 1 : 0} onClick={() => act("change_hair_portion", { "new_portion": "bottom" })}>Bottom Hair</Button.Checkbox>
+      <Button.Checkbox checked={selected_hair_portion === "middle" ? 1 : 0} onClick={() => act("change_hair_portion", { "new_portion": "middle" })}>Middle Hair</Button.Checkbox>
+      <Button.Checkbox checked={selected_hair_portion === "top" ? 1 : 0} onClick={() => act("change_hair_portion", { "new_portion": "top" })}>Top Hair</Button.Checkbox>
+      <hr />
+      <Button bold color="red" icon="cut" onClick={() => act("do_hair", { "style_id": null })}>Create Wig</Button>
+    </Stack>
+  );
+};
+
+const HairPreview = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { hair_style, hair_name } = props;
+  return (
+    <Section width="140px" direction="column" align="center">
+      <Image pixelated width="60px" height="100px" src={`${hair_style["hair_icon"]}`} />
+      <Box width="100%" fontSize="15px" textAlign="center" mb="10px">{hair_name}</Box>
+      <Stack inline justify="space-between">
+        <Button color="blue" height="20px" icon="cut" onClick={() => act("do_hair", { "style_id": hair_style["hair_id"] })}>Cut</Button>
+        <Button color="blue" height="20px" icon="eye" onClick={() => act("update_preview", { "action": "new_hair", "style_id": hair_style["hair_id"] })}>Preview</Button>
+      </Stack>
+    </Section>
+  );
+};
+
+const PreviewWindow = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { preview, current_hair_style } = data;
+  return (
+    <Section>
+
+      <Stack justify="space-around" height="200px" wrap="wrap">
+        <Button icon="rotate-left" color="red" height="22px" width="100%" onClick={() => act("update_preview", { "action": "reset" })}>Reset</Button>
+        <ByondUi
+          params={{
+            id: preview,
+            type: "map",
+          }}
+          style={{
+            width: "80px",
+            height: "160px",
+          }} />
+      </Stack>
+      <hr />
+      <Stack wrap="wrap" height="100px" align="center">
+        <Box mb="2px" width="100%" bold>Top Hair: {current_hair_style["top"]}</Box>
+        <Box mb="2px" width="100%" bold>Middle Hair: {current_hair_style["middle"]}</Box>
+        <Box mb="2px" width="100%" bold>Bottom Hair: {current_hair_style["bottom"]}</Box>
+      </Stack>
```

Comment:
Apply `mb` to `Stack` rather than individual `Box` elements? Also sounds like you *might* want the `Section` itself to have a `Stack` as its immediate child, possibly? Should save some of the manual styling being done.

Reply:
It wasn't working for whatever reason, but after nesting it under a `<Stack.Item>`, it looked neat enough to not need any margins. Here's how it looks right now:
![Screenshot_460](https://user-images.githubusercontent.com/36608862/213632304-3a523020-f1f2-4972-ba3f-5ddb630091f8.png)